File: minipack-core/src/test/java/io/github/odenix/minipack/core/example/WriteArray.java
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.github.odenix.minipack.core.example;

import net.jqwik.api.ForAll;
import io.github.odenix.minipack.core.MessageWriter;

import java.io.IOException;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class WriteArray extends Example {
  // -8<- [start:snippet]
  void write(MessageWriter writer, List<String> list) throws IOException {
    writer.writeArrayHeader(list.size());
    for (var element : list) {
      writer.write(element);
    }
  }
  // -8<- [end:snippet]

  @net.jqwik.api.Example
  void test(@ForAll List<String> list) throws IOException {
    write(writer, list);
    writer.close();
    assertThat(new ReadArray().read(reader)).isEqualTo(list);
  }
}

File: minipack-kotlin/src/test/kotlin/io/github/odenix/minipack/kotlin/example/WriteArray.kt
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.github.odenix.minipack.kotlin.example

import io.github.odenix.minipack.core.MessageWriter
import net.jqwik.api.ForAll
import org.assertj.core.api.Assertions.assertThat

class WriteArray : Example() {
  // -8<- [start:snippet]
  fun write(writer: MessageWriter, list: List<String>) {
    writer.writeArrayHeader(list.size)
    for (element in list) {
      writer.write(element)
    }
  }
  // -8<- [end:snippet]

  @net.jqwik.api.Example
  fun test(@ForAll list: List<String>) {
    write(writer, list)
    writer.close()
    assertThat(ReadArray().read(reader)).isEqualTo(list)
  }
}
